### DISCUSSION TITLE
Robot accessibility

### DISCUSSION THREAD

**Discussion by xPhilippeRoux** [HALRobotics/Beta#92] [Ideas]
Do you plan on adding a possibility to provide the reachable envelope of a specific mechanism automatically ?

Thanks.

Philippe

**Comment by sebandraos** [comment]
Hi @XtreeE-PRO ,
Do you mean some form of graphical representation of the reachable envelope (like the image below in 3D) or something a bit more involved that would take into account Target orientations etc.?

![image](https://user-images.githubusercontent.com/6084391/112629714-8ff78280-8e2c-11eb-80d3-71bc39833949.png)

**Reply by xPhilippeRoux** [reply]
Exactly this in 3D, so that you are sure that if a target is inside this envelope, then there exists at least one configuration for the overall mechanism to reach it.

Then of course it depends the application and the tool. For my application, what defines my target is only a point + a normal vector, so that frames can freely rotate around themselves (as my tool has one main direction only). For that I would be happy to be able to use the extended envelope, where if a target is inside, then there exists at least one rotation around the target (same point, same normal vector) that the overall mechanism can reach.

Hope it's clear enough.

**Reply by sebandraos** [reply]
Ok, I think this could be a component that we add to the Grasshopper Client so we can include things like the End Effector, Target orientations and composite Mechanisms. I'll have a think with the rest of the team and see what we can come up with.
